### PR TITLE
fix: use ResizeObserver API  for container size changes

### PR DESCRIPTION
**Pull request by jikkai**
Replace the resize event with the **ResizeObserver API**, as the container's size may depend on factors beyond just the window size, such as changes in other DOM elements like opening & closing a sidebar.

### CODE REVIEW

**Comment by Jocs** [comment]
Great!

File: packages/base-render/src/Engine.ts [hunk 1]
```diff
@@ -122,11 +121,21 @@ export class Engine extends ThinEngine<Scene> {
         this._container.appendChild(this._canvasEle);
 
         this.resize();
-        this.disposeWithMe(fromWindowEvent('resize', () => this.resize()));
+        const resizeObserver = new ResizeObserver(() => {
+            window.requestIdleCallback(() => {
+                this.resize();
```
Comment (wzhudev) [inline packages/base-render/src/Engine.ts:126]: 这里是否需要做一个 debounce 或者 throttle 操作？

File: packages/base-render/src/Engine.ts [hunk 2]
```diff
@@ -122,11 +121,28 @@ export class Engine extends ThinEngine<Scene> {
         this._container.appendChild(this._canvasEle);
 
         this.resize();
-        this.disposeWithMe(fromWindowEvent('resize', () => this.resize()));
+
+        let timer: number | undefined;
+        const resizeObserver = new ResizeObserver(() => {
+            if (!timer) {
+                timer = window.setTimeout(() => {
+                    window.requestIdleCallback(() => {
```
Comment (wzhudev) [inline packages/base-render/src/Engine.ts:129]: requestIdleCallback 也会返回一个 number 用于取消 timer，因此无需用 setTimeout 来做调度

Comment (jikkai) [reply packages/base-render/src/Engine.ts:129]: 👌

**Review by wzhudev — APPROVED** [review]

**Pull request merged**